### ISSUE TITLE
Update CI to use pyiron/actions

### ISSUE THREAD

**Issue by liamhuber** [pyiron/pyiron_atomistics#861]
Analogous to how [pyiron/ironflow](https://github.com/pyiron/ironflow) now leans on reusable workflows from [pyiron/actions](https://github.com/pyiron/actions), although here (a) we have some workflows that are not yet covered by the central CI repo, and (b) the primary branch here is named `master` not `main` (for now! But this change will actually be easier after we lean on the centralized repo, as it gives us fewer locations where we'll need to update the name in our source files...)

Steps, as best as I see them now:
- [ ] Add new CI files to `.github/workflows` just like in ironflow (but targeting master instead of main) to call each of the reusable workflows
- [ ] Remove CI files that are now redundant
- [ ] Remove `.ci_support` content and other material that is now redundant because it's called from `pyiron/actions/.support` instead
- [ ] Look at the other pyiron repositories and see if any remaining workflows can be abstracted over to pyiron/actions
- [ ] If yes, get them re-written in a more abstract way over there and then repeat the first three steps
- [ ] If no, at least see if some of the workflow content can be replaced by calls to actions in pyiron/actions, and repeat the third step

Finally, we'll need to go repeat this for the other pyiron repositories.

**Comment by liamhuber** [comment]
I did this on the pyiron_module_template in [this PR](https://github.com/pyiron/pyiron_module_template/pull/10) -- although be warned other stuff got done here too (e.g. black formatting that touched a tonne of files), so I'm sorry it's not the easiest to read!

**Comment by liamhuber** [comment]
Bump @mbruns91. @srmnitc, do I remember correctly that you were going to look at the same process for another repo? I thought maybe it was base, but don't see a sister issue there.

**Comment by mbruns91** [comment]
I started reviewing [mentioned PR](https://github.com/pyiron/pyiron_module_template/pull/10) (though it's already merged... Still a good example for me).
I'm super sorry this has taken so freaking long. The end of last year was just way to stressful and I had to put my focus on other stuff. I will work on applying things to atomistics this week.

**Comment by srmnitc** [comment]
@liamhuber Yes, sorry it took a bit. I am on sick leave now, I'll definitely work on this when I am back next week.

**Comment by liamhuber** [comment]
No worries, it's not burning urgent, I just wanted to make sure it's on the radar.

> I am on sick leave now

Gute Besserung!!!
